File: android/ac-teams/src/main/kotlin/com/microsoft/adaptivecards/teams/DeepLinkHandler.kt
package com.microsoft.adaptivecards.teams

import android.net.Uri

class DeepLinkHandler {
    
    fun parseDeepLink(uri: Uri): DeepLinkInfo? {
        if (uri.scheme != "msteams") return null
        
        val parameters = mutableMapOf<String, String>()
        uri.queryParameterNames.forEach { name ->
            uri.getQueryParameter(name)?.let { value ->
                parameters[name] = value
            }
        }
        
        return DeepLinkInfo(
            scheme = uri.scheme ?: "",
            host = uri.host ?: "",
            path = uri.path ?: "",
            parameters = parameters
        )
    }
    
    fun handleNavigation(deepLink: DeepLinkInfo) {
        // Host app implements actual navigation
    }
}

File: ios/Sources/ACTeams/DeepLinkHandler.swift
import Foundation

public class DeepLinkHandler {
    public init() {}
    
    public func parseDeepLink(_ url: URL) -> DeepLinkInfo? {
        guard url.scheme == "msteams" else { return nil }
        
        var parameters: [String: String] = [:]
        if let components = URLComponents(url: url, resolvingAgainstBaseURL: false),
           let queryItems = components.queryItems {
            for item in queryItems {
                if let value = item.value {
                    parameters[item.name] = value
                }
            }
        }
        
        return DeepLinkInfo(
            scheme: url.scheme ?? "",
            host: url.host ?? "",
            path: url.path,
            parameters: parameters
        )
    }
    
    public func handleNavigation(_ deepLink: DeepLinkInfo) {
        // Host app implements actual navigation
    }
}
